### PR TITLE
Add null pointer exception to fix the crash

### PR DESCRIPTION
**Pull request by abhishek1508**
## Description

Fix Null Pointer Exception [#2555](https://github.com/mapbox/mapbox-navigation-android/issues/2555)

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### Goal

Fix Null Pointer Exception

### Implementation

Fix Null Pointer Exception

## Screenshots or Gifs

Please include all the media files to give some context about what's being implemented or fixed. It's not mandatory to upload screenshots or gifs, but for most of the cases it becomes really handy to get into the scope of the feature / bug being fixed and also it's REALLY useful for UI related PRs screenshot gif

## Testing

Please describe the manual tests that you ran to verify your changes

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionModel.java [hunk 1]
```diff
@@ -12,10 +15,21 @@
   private String drivingSide;
 
   public InstructionModel(DistanceFormatter distanceFormatter, RouteProgress progress) {
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionModel.java]: As this is a `public` method, could we add `@Nullable` / `@NonNull` annotations to help Kotlin users?

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionModel.java [hunk 2]
```diff
@@ -12,10 +15,21 @@
...
-    this.progress = progress;
-    double distanceRemaining = progress.currentLegProgress().currentStepProgress().distanceRemaining();
-    stepDistanceRemaining = distanceFormatter.formatDistance(distanceRemaining);
-    this.drivingSide = progress.currentLegProgress().currentStepProgress().step().drivingSide();
+    if (progress != null && distanceFormatter != null) {
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionModel.java]: Could we simplify this `if` nested chain somehow? I'm wondering why we didn't run into this 💥 before in the legacy 🤔 What did it change recently?

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/summary/list/InstructionListPresenter.java [hunk 3]
```diff
@@ -115,12 +115,16 @@ private void addBannerInstructions(RouteProgress routeProgress) {
     if (isNewLeg(routeProgress)) {
       instructions = new ArrayList<>();
       currentLeg = routeProgress.currentLegProgress().routeLeg();
-      drivingSide = routeProgress.currentLegProgress().currentStepProgress().step().drivingSide();
-      List<LegStep> steps = currentLeg.steps();
-      for (LegStep step : steps) {
-        List<BannerInstructions> bannerInstructions = step.bannerInstructions();
-        if (bannerInstructions != null && !bannerInstructions.isEmpty()) {
-          instructions.addAll(bannerInstructions);
+      if (routeProgress.currentLegProgress().currentStepProgress().step() != null) {
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/summary/list/InstructionListPresenter.java]: Same as above re: wondering why we haven't run into this NPE before 🤔

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionModel.java (hunk 2 above)
Comment (abhishek1508) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionModel.java]: In the core SDK, RouteProgress and every object inside it @Nullable. This was not the case earlier. Plus we are trying to start navigation without a route which was not the case earlier.

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/summary/list/InstructionListPresenter.java (hunk 3 above)
Comment (abhishek1508) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/summary/list/InstructionListPresenter.java]: In the core SDK, RouteProgress and every object inside it @Nullable. This was not the case earlier. Plus we are trying to start navigation without a route which was not the case earlier.

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionModel.java (hunk 1 above)
Comment (abhishek1508) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionModel.java]: I think it should be `@Nullable`. Considering the Free Drive mode, routeProgress can be null.

Comment (Guardiola31337) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/instruction/InstructionModel.java]: What about `DistanceFormatter`?

**Pull request merged**